shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-sqlserver/src/main/java/org/apache/shardingsphere/sql/parser/sqlserver/visitor/statement/impl/SQLServerDMLStatementSQLVisitor.java

```diff
@@ -243,7 +243,7 @@ public ASTNode visitWithClause(final WithClauseContext ctx) {
         List<CteClauseContext> cteClauses = ctx.cteClause();
         Collection<CommonTableExpressionSegment> commonTableExpressions = new LinkedList<>();
         for (CteClauseContext cte : cteClauses) {
-            SubquerySegment subquery = new SubquerySegment(cte.start.getStartIndex(), cte.stop.getStopIndex(), (SQLServerSelectStatement) visit(cte.subquery()));
+            SubquerySegment subquery = new SubquerySegment(cte.subquery().start.getStartIndex(), cte.subquery().stop.getStopIndex(), (SQLServerSelectStatement) visit(cte.subquery()));
```

Comment:
Why do you modify the SQLServer visitor?
As this pr is only for Oracle.

Reply:
While I was adding start and stop indices for all `subquery-expression` within `common-table-expression`, in the SQLServer visitor, the `SubquerySegment`'s start and stop indices are collected from `CommonTableExpressionSegment`'s start and stop indices. So I changed them from getting `CommonTableExpressionSegment`'s `SubquerySegment`. Shall I remove this modification?